File: app/src/main/java/com/kevinmost/leddit/prefs/BooleanPreference.java
package com.kevinmost.leddit.prefs;

import javax.inject.Inject;

public enum BooleanPreference implements LedditPreference<Boolean> {

    ;

    @Inject
    LedditPreferences prefs;

    private final String key;
    private final Boolean defaultValue;

    private BooleanPreference(String key) {
        this(key, false);
    }

    private BooleanPreference(String key, Boolean defaultValue) {
        this.key = key;
        this.defaultValue = defaultValue;
    }

    @Override
    public Boolean getValue() {
        return prefs.get(this);
    }

    @Override
    public void setValue(Boolean value) {
        prefs.put(this, value);
    }

    @Override
    public String getKey() {
        return key;
    }

    @Override
    public Boolean getDefaultValue() {
        return defaultValue;
    }
}

File: app/src/main/kotlin/com/kevinmost/leddit/preference/BooleanPreference.kt
package com.kevinmost.leddit.preference

import android.content.SharedPreferences
import javax.inject.Inject

public enum class BooleanPreference private(key: String, defaultValue: Boolean = false):
        LedditPreference<Boolean> {

    FOO: BooleanPreference("bar");

    override protected var prefs: SharedPreferences? = null;
    override protected val key: String = key;
    override protected val defaultValue: Boolean = defaultValue;

    override fun SharedPreferences.Editor.put(value: Boolean): SharedPreferences.Editor {
        putBoolean(key, value);
        return this;
    }

    override fun SharedPreferences.get(): Boolean {
        return getBoolean(key, defaultValue);
    }
}